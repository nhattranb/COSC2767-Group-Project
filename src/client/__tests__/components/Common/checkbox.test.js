/*
 *   RMIT University Vietnam
 *   Course: COSC2767 Systems Deployment and Operations
 *   Semester: 2024C
 *   Assessment: Assignment 2
 *   Author: Le Dinh Ngoc Quynh
 *   ID: s3791159
 *   Created  date: 17 Dec 2024
 *   Last modified: 17 Dec 2024
 */

import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Checkbox from '../../../app/components/Common/Checkbox';
import {describe, expect, it} from "@jest/globals";

describe('Checkbox component', () => {
    it('renders correctly with default props', () => {
        const { asFragment } = render(<Checkbox id="test-checkbox" name="test" />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly with custom className', () => {
        const { asFragment } = render(<Checkbox id="test-checkbox" name="test" className="custom-class" />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly with label', () => {
        const { asFragment } = render(<Checkbox id="test-checkbox" name="test" label="Test Label" />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly when checked', () => {
        const { asFragment } = render(<Checkbox id="test-checkbox" name="test" checked />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly when disabled', () => {
        const { asFragment } = render(<Checkbox id="test-checkbox" name="test" disabled />);
        expect(asFragment()).toMatchSnapshot();
    });
});
