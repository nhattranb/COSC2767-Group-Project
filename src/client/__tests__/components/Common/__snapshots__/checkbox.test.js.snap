// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component renders correctly when checked 1`] = `
<DocumentFragment>
  <div
    class="checkbox custom-icon "
  >
    <input
      checked=""
      class="input-checkbox"
      id="test-checkbox"
      name="test"
      type="checkbox"
    />
    <label
      for="test-checkbox"
      type="submit"
    />
  </div>
</DocumentFragment>
`;

exports[`Checkbox component renders correctly when disabled 1`] = `
<DocumentFragment>
  <div
    class="checkbox custom-icon "
  >
    <input
      class="input-checkbox"
      id="test-checkbox"
      name="test"
      type="checkbox"
    />
    <label
      for="test-checkbox"
      type="submit"
    />
  </div>
</DocumentFragment>
`;

exports[`Checkbox component renders correctly with custom className 1`] = `
<DocumentFragment>
  <div
    class="checkbox custom-icon custom-class"
  >
    <input
      class="input-checkbox"
      id="test-checkbox"
      name="test"
      type="checkbox"
    />
    <label
      for="test-checkbox"
      type="submit"
    />
  </div>
</DocumentFragment>
`;

exports[`Checkbox component renders correctly with default props 1`] = `
<DocumentFragment>
  <div
    class="checkbox custom-icon "
  >
    <input
      class="input-checkbox"
      id="test-checkbox"
      name="test"
      type="checkbox"
    />
    <label
      for="test-checkbox"
      type="submit"
    />
  </div>
</DocumentFragment>
`;

exports[`Checkbox component renders correctly with label 1`] = `
<DocumentFragment>
  <div
    class="checkbox default-icon "
  >
    <input
      class="input-checkbox"
      id="test-checkbox"
      name="test"
      type="checkbox"
    />
    <label
      for="test-checkbox"
      type="submit"
    >
      Test Label
    </label>
  </div>
</DocumentFragment>
`;
